shims/src/peripherals/gpio.rs

```diff
@@ -156,14 +161,35 @@ impl<'a> Gpio<'a> for GpioShim<'a> {
         }
     }
 
-    fn register_interrupt(
-        &mut self,
-        pin: GpioPin,
-        handler: GpioHandler<'a>,
-    ) -> Result<(), GpioMiscError> {
-        self.handlers[pin] = handler;
+    // TODO: decide functionality when no previous flag registered
```

Comment:
when flags aren't encountered and we hit an interrupt we should panic

It's in invariant we'll enforce manually (there's builder pattern stuff in the `InstructionInterpreter` and above to make sure we can't actually do this), but just in case we mess up we should make it very obvious that something is wrong.

`panic!()` or `unreachable!()`, etc.

Reply:
Whoops this was meant to say "decide functionality when previous flag IS registered. Overwrite or panic?